src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -452,9 +452,20 @@ public static Map<String, DetailNode> getPropertiesJavadocs(Set<String> properti
         final Map<String, DetailNode> unmodifiableJavadocs =
                 ClassAndPropertiesSettersJavadocScraper.getJavadocsForModuleOrProperty();
         final Map<String, DetailNode> javadocs = new LinkedHashMap<>(unmodifiableJavadocs);
-        properties.forEach(property -> {
+
+        for (String property : properties) {
             javadocs.putIfAbsent(property, SUPER_CLASS_PROPERTIES_JAVADOCS.get(property));
-        });
+
+            final boolean isPropertySetterJavadocNotFound = javadocs.get(property) == null
+                        && !TOKENS.equals(property) && !JAVADOC_TOKENS.equals(property);
```

Comment:
I will be surprised if something doesn't flag this, there should not be a negative in a boolean name.
```suggestion
            final boolean isPropertySetterJavadocFound = javadocs.get(property) != null
                       || TOKENS.equals(property) || JAVADOC_TOKENS.equals(property);
```

Reply:
Either way: @stoyanK7, do you agree that proposed change is easier to understand and reason about?

Reply:
Or, even better:
```suggestion
            final boolean isPropertySetterJavadocFound = javadocs.containsKey(property) 
            || TOKENS.equals(property) || JAVADOC_TOKENS.equals(property);
```

I didn't realize this was a map for a second, since we didn't have name like `valueForKey` :)

Reply:
> Or, even better:

Sounds good. What about this additional check on the top?
```java
            if (!javadocs.containsKey(property)
                    && SUPER_CLASS_PROPERTIES_JAVADOCS.containsKey(property)) {
                javadocs.put(property, SUPER_CLASS_PROPERTIES_JAVADOCS.get(property));
            }

            final boolean isPropertySetterJavadocFound = javadocs.containsKey(property)
                       || TOKENS.equals(property) || JAVADOC_TOKENS.equals(property);
            if (!isPropertySetterJavadocFound) {
```

Without it `javadocs.containsKey(property) ` would still return `true` with the value being `null`

Reply:
Ah yes, I like this, makes this somewhat hidden condition more obvious

Reply:
Done.